src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -1152,6 +1159,7 @@ private async Task<T> ExecuteWithFailOverPolicyAsync<T>(
                     if (!IsFailOverable(ae) || !clientEnumerator.MoveNext())
                     {
                         backoffAllClients = true;
```

Comment:
It looks like this should be set in the finally block inside the `if (!success...)` check.